### PR TITLE
Removes floor from flip message when falling over in space

### LINKED ISSUE #24471: *flip in space summons invisible floors
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1623

### Issue Description

If you use *flip in space, you can trigger these messages:
* "attempts a flip and crashes to the floor!"  (5% chance normally)
* "flops and flails around on the floor." (If already horizontal)

### What did you expect to happen?

There is no floor in space.

### What happened instead?

Magic space floor.

### Why is this bad/What are the consequences?

It's weird.

### Steps to reproduce the issue.

1. Go into space.
2. *flip
3. *flip
4. [...]
5. *flip and crash to the floor.
6. *flip and flail around on the floor.

### When did the problem start happening?

Unknown

### Extra information

Code is here:
https://github.com/ParadiseSS13/Paradise/blob/9a1e46b9690c1c4f84c84406beb64137d83c3e4a/code/modules/mob/mob_emote.dm#L128

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by Social-Moth**
## What Does This PR Do

Adds a message for failing a flip while on a space tile.

## Why It's Good For The Game

There is no floor in space, so crashing into the floor shouldn't happen there.

resolves #24471 

## Testing

Compile, spam *flip until it fails on station, then repeat in space. Works as intended in both scenario's

## Changelog
:cl:
tweak: Changed text for falling over during a flip while in space
/:cl:

### CODE REVIEW

File: code/modules/mob/mob_emote.dm [hunk 1]
```diff
@@ -134,8 +134,12 @@
 	var/mob/living/L = user
 
 	if(IS_HORIZONTAL(L))
+		var/turf = get_turf(L)
+		if(isspaceturf(turf))
+			message = "flops and flails around in space"
+		else
 		message = "flops and flails around on the floor."
```
Comment (Contrabang) [inline code/modules/mob/mob_emote.dm:141]:
```suggestion
			message = "flops and flails around on the floor."
```

File: code/modules/mob/mob_emote.dm [hunk 2]
```diff
@@ -134,8 +134,12 @@
...
-		return ..()
+			return ..()
```
Comment (Contrabang) [inline code/modules/mob/mob_emote.dm:142]:
```suggestion
		return ..()
```

File: code/modules/mob/mob_emote.dm [hunk 3]
```diff
@@ -166,11 +170,15 @@
...
-		message = "attempts a flip and crashes to the floor!"
-		sleep(0.3 SECONDS)
+		var/turf = get_turf(L)
+		if(isspaceturf(turf))
+			message = "attempts a flip and loses balance!"
+		else
+			message = "attempts a flip and crashes to the floor!"
 		if(istype(L))
+			sleep(0.3 SECONDS)
 			L.Weaken(4 SECONDS)
```
Comment (Contrabang) [inline code/modules/mob/mob_emote.dm:180]: We're trying to get rid of sleeps, you can instead use a timer here. This should work the same
```suggestion
			addtimer(CALLBACK(L, TYPE_PROC_REF(/mob/living, Weaken), 4 SECONDS), 0.3 SECONDS)
```

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/modules/mob/mob_emote.dm [hunk 4]
```diff
@@ -166,11 +170,14 @@
...
+		var/turf = get_turf(L)
+		if(isspaceturf(turf))
+			message = "attempts a flip and loses balance!"
+		else
+			message = "attempts a flip and crashes to the floor!"
 		if(istype(L))
-			L.Weaken(4 SECONDS)
-		return ..()
+			addtimer(CALLBACK(L, TYPE_PROC_REF(/mob/living, Weaken), 4 SECONDS), 0.3 SECONDS)
+			return ..()
```
Comment (Contrabang) [inline code/modules/mob/mob_emote.dm:180]: I fucked this up, my bad
```suggestion
		return ..()
```

File: code/modules/mob/mob_emote.dm [hunk 5]
```diff
@@ -134,7 +134,11 @@
 	var/mob/living/L = user
 
 	if(IS_HORIZONTAL(L))
-		message = "flops and flails around on the floor."
+		var/turf = get_turf(L)
+		if(isspaceturf(turf))
+			message = "flops and flails around in space"
```
Comment (Henri215) [inline code/modules/mob/mob_emote.dm:139]:
```suggestion
			message = "flops and flails around in space."
```

File: code/modules/mob/mob_emote.dm [hunk 6]
```diff
@@ -166,10 +170,13 @@
 							"<span class='warning'>As you flip your hat falls off!</span>")
 
 	if(prob(5) && ishuman(user))
-		message = "attempts a flip and crashes to the floor!"
-		sleep(0.3 SECONDS)
+		var/turf = get_turf(L)
+		if(isspaceturf(turf))
+			message = "attempts a flip and loses balance!"
+		else
+			message = "attempts a flip and crashes to the floor!"
```
Comment (DGamerL) [inline code/modules/mob/mob_emote.dm:177]:
```suggestion
		message = "attempts a flip and [isspaceturf(turf) ? "loses balance" : "crashes to the floor"]!"
```

File: code/modules/mob/mob_emote.dm [hunk 7]
```diff
@@ -134,7 +134,11 @@
 	var/mob/living/L = user
 
 	if(IS_HORIZONTAL(L))
-		message = "flops and flails around on the floor."
+		var/turf = get_turf(L)
+		if(isspaceturf(turf))
+			message = "flops and flails around in space"
+		else
+			message = "flops and flails around on the floor."
```
Comment (DGamerL) [inline code/modules/mob/mob_emote.dm:141]:
```suggestion
		message = "flops and flails around [isspaceturf(turf) ? "in space" : "on the floor"]."
```

**Review by DGamerL — COMMENTED** [review]
Ternaries my beloved

**Review by DGamerL — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Review by Contrabang — APPROVED** [review]

**Pull request merged**